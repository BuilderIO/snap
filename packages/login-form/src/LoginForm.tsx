import { Input } from '@taxi/input';
import * as React from 'react';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles({
  root: {
    display: 'flex',
    flexDirection: 'column',
    margin: '10px',
  },
  wrapper: {
    textAlign: 'right',
  },
});

export interface LoginFormProps {
  onClick?: () => void;
}

export const LoginForm: React.FunctionComponent<LoginFormProps> = ({
  onClick,
}) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Input id="name" label="Name" />
      <Input id="password" label="Password" />

      <div className={classes.wrapper}>
        <button onClick={onClick}>Log in</button>
      </div>
    </div>
  );
};
