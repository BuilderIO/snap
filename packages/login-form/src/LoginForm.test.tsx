import React from 'react';
import { render } from '@testing-library/react';

import { LoginForm } from './LoginForm';

jest.mock('@taxi/input', () => ({ Input: 'input' }));

describe('LoginForm', () => {
  test('should match snapshot and styles', () => {
    expect(render(<LoginForm />).container).toMatchSnapshot();
  });
});
