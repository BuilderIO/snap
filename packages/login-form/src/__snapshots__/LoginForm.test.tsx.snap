// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match snapshot and styles 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <input
      id="name"
      label="Name"
    />
    <input
      id="password"
      label="Password"
    />
    <div
      class="makeStyles-wrapper-2"
    >
      <button>
        Log in
      </button>
    </div>
  </div>
</div>
`;
